pkg/proxy/iptables/proxier_test.go

```diff
@@ -1121,4 +1121,44 @@ func TestBuildServiceMapServiceHeadless(t *testing.T) {
 	}
 }
 
+func TestBuildServiceMapServiceTypeExternalName(t *testing.T) {
+	services := []api.Service{
+		{
+			ObjectMeta: api.ObjectMeta{
```

Comment:
For this test and the other's like it, it might be really nice to avoid boilerplate clutter and do something like:

```
services := []api.Service{
    makeService("external-name", func(svc *api.Service) {
        svc.Spec.Type = api.ServiceTypeExternalName
        svc.Spec.ExternalName = "foo2.bar.com"
        svc.Spec.ClusterIP = "172.16.55.4"
    })
```

That way the boilerplate struct gunk is the same (or randomized) and the testcase ONLY carries the things that are unique, which makes it easier to comprehend each case.